### PR TITLE
update readstatus lastUpdated when users interact with comments in subscribed feed

### PR DESCRIPTION
**Pull request by b0b3rt**
This reintroduces functionality somewhat similar to what was removed in https://github.com/ForumMagnum/ForumMagnum/pull/6309, except only for the Subscribed feed, and if there isn't already a ReadStatus for a post we create it with `isRead: false` (since in the case where a user e.g. clicks on a comment on a feed item for a post that they haven't read yet, it'd be wrong to set `isRead: true`, while updating an existing ReadStatus should avoid changing the `isRead` value).
┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1207556787974795) by [Unito](https://www.unito.io)

### CODE REVIEW

File: packages/lesswrong/components/hooks/useRecordPostView.tsx [hunk 1]
```diff
@@ -112,8 +120,25 @@ export const useRecordPostView = (post: ViewablePost) => {
...
+  const recordPostCommentsView = ({ post }: Pick<RecordPostViewArgs, 'post'>) => {
+    if (currentUser) {
+      if (!postsRead[post._id]) {
+        // Update the client-side read status cache.
+        // Set the value to true even if technically we might be saving isRead: false on the server, if the post hasn't been read before, to get the correct UI update.
+        setPostRead(post._id, true);
+
+        // Only send update to server the first time a user interacts with the comments after the page is rendered (and the client-side cache isn't populated yet).
+        // Otherwise it becomes much more likely that someone else posts a comment after the first time we send this,
+        // but then we send it again because e.g. the user clicked on another comment (from the initial render).
```
Comment (jpaddison3) [inline packages/lesswrong/components/hooks/useRecordPostView.tsx:133]: I'm not seeing the logic you have that achieves this.

File: packages/lesswrong/components/recentDiscussion/FeedPostCommentsCard.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import React, { useState } from 'react';
+import React, { useRef, useState } from 'react';
```
Comment (jpaddison3) [inline packages/lesswrong/components/recentDiscussion/FeedPostCommentsCard.tsx:1]:
```suggestion
import React, { useState } from 'react';
```

**Review by jpaddison3 — APPROVED** [review]

**Comment by jpaddison3** [comment]
Ok, so just so I understand, the desired behavior is:

If I _have_ read the post, and I click on the comment thread in FeedPostCommentsCard:
* Update the read status to mark the most recent read and reset the clock for the unread markers

If I have _not_ read the post:
* Insert a read status with an isRead=FALSE which provides some benefit maybe, but crucially does not turn the postsitems grey

File: packages/lesswrong/components/hooks/useRecordPostView.tsx (hunk 1 above)
Comment (b0b3rt) [reply packages/lesswrong/components/hooks/useRecordPostView.tsx:133]: It being in the if-block at all depends on `postsRead` not having been populated for that post id yet, which happens on L129 (and so happens the first time `recordPostCommentsView` is called).  I'll try to clarify the comment

**Comment by b0b3rt** [comment]
> Ok, so just so I understand, the desired behavior is:
> 
> If I _have_ read the post, and I click on the comment thread in FeedPostCommentsCard:
> 
> * Update the read status to mark the most recent read and reset the clock for the unread markers
> 
> If I have _not_ read the post:
> 
> * Insert a read status with an isRead=FALSE which provides some benefit maybe, but crucially does not turn the postsitems grey

Correct, that's the intended behavior.  `FeedPostCommentsCard` is the feed item in our new Subscribed feed.

File: packages/lesswrong/components/recentDiscussion/FeedPostCommentsCard.tsx (hunk 2 above)
Comment (b0b3rt) [reply packages/lesswrong/components/recentDiscussion/FeedPostCommentsCard.tsx]: 👍

**Pull request merged**